packages/react-router/src/preloader.tsx

```diff
@@ -47,6 +50,24 @@ export function createApolloLoaderHandler(
   };
 }
 
+
+export const apolloContext = unstable_createContext<ApolloContext>();
+
+export function createApolloMiddleware(
+  makeClient: (request: Request) => ApolloClient
+): unstable_MiddlewareFunction {
+  return async ({ request, context }) => {
+    const client = makeClient(request);
+    const preloader = createTransportedQueryPreloader(client);
+    const preloadQuery: typeof preloader = (...args) =>
+      replaceStreamWithPromiscade(preloader(...args));
```

Comment:
Since we're working in the browser here, and don't need to transport data from the server to the browser, it would be optimal if we could somehow use the same `client` instance that the React tree sees and use the normal `createQueryPreloader` of Apollo Client - see https://www.apollographql.com/docs/react/data/suspense#initiating-queries-outside-react

Reply:
Oh no, I mistook this as a feature only for client loaders. We might be getting some symmetry between server and client out of this in the end. Let's see.